lib/matplotlib/tests/test_container.py

```diff
@@ -28,3 +28,20 @@ def test_errorbar_remove():
 
     eb = ax.errorbar([1], [1], fmt='none')
     eb.remove()
+
+def test_removal_from_legend():
+    ax = plt.gca()
+
+    obj1 = ax.errorbar([1], [1], [1], marker="o", label="foo")
+
+    obj1.remove()
+
+    ax.errorbar([1.1], [1.1], [1], marker="o", label="bar")
+
+    plt.legend()
```

Comment:
This line is not needed I think.

Also, stylistically it would be good to remove some of the blank lines here.  Try to use blank lines to separate things into related blocks of code.  So maybe the lines above this (setting up) could be one block, and the lines below (the actual checks) could be another.

Functionally, this test seems good to me 👍